lib/baby_squeel/join_dependency.rb

```diff
@@ -20,6 +20,29 @@ def group_by
       end
     end
 
+    # This class allows BabySqueel to slip custom
+    # joins_values into Active Record's JoinDependency
+    class Injector6_0 < Array # :nodoc:
+      # https://github.com/rails/rails/pull/36805/files
+      # This commit changed group_by to each
+      def each(&block)
```

Comment:
Hmm, okay. We might want to consider doing that too, especially since you said the current approach won't work for 6.1. I'm going to merge this for now, but something worth thinking about.